### PR TITLE
Update everything and cleanup

### PR DESCRIPTION
**Pull request by SUPERCILEX**

### CODE REVIEW

File: local.properties [hunk 1]
```diff
@@ -1,10 +0,0 @@
-## This file is automatically generated by Android Studio.
-# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
-#
-# This file should *NOT* be checked into Version Control Systems,
-# as it contains information specific to your local configuration.
-#
-# Location of the SDK. This is only used by Gradle.
-# For customization when using a Version Control System, please read the
-# header note.
-sdk.dir=/opt/android/android-sdk-linux
```
Comment (SUPERCILEX) [inline local.properties]: @centic9 You'll have to re-add this to your project (it's now excluded from git as it should be).

File: poitest/proguard-rules.pro [hunk 2]
```diff
@@ -1,17 +1,3 @@
...
-
-# If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
-# class:
-#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
-#   public *;
-#}
+-dontwarn org.apache.**
+-dontwarn org.openxmlformats.**
+-dontwarn com.fasterxml.aalto.stax.**
```
Comment (SUPERCILEX) [inline poitest/proguard-rules.pro]: This fixes release build errors.

File: poitest/src/main/java/org/dstadler/poiandroidtest/poitest/DocumentDetailFragment.java [hunk 3]
```diff
@@ -26,13 +27,6 @@
      */
     private DummyContent.DummyItem mItem;
 
-    /**
-     * Mandatory empty constructor for the fragment manager to instantiate the
-     * fragment (e.g. upon screen orientation changes).
-     */
-    public DocumentDetailFragment() {
-    }
```
Comment (SUPERCILEX) [inline poitest/src/main/java/org/dstadler/poiandroidtest/poitest/DocumentDetailFragment.java]: You just shouldn't be declaring constructors at all.

File: poitest/src/main/java/org/dstadler/poiandroidtest/poitest/DocumentListFragment.java [hunk 4]
```diff
@@ -89,15 +83,15 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
     }
 
     @Override
-    public void onAttach(Activity activity) {
```
Comment (SUPERCILEX) [inline poitest/src/main/java/org/dstadler/poiandroidtest/poitest/DocumentListFragment.java]: Deprecated.

**Comment by SUPERCILEX** [comment]
@centic9 what do you think?

**Comment by centic9** [comment]
FYI, I started to apply some of the changes manually as just applying the large list contained here did not seem a good idea as they are quite varied and thus hard to keep apart. 

Also some thing did not work for me, e.g. excluding the .xsb files breaks the test application itself, did you test the resulting apk with the updated shadow-jar?

**Comment by centic9** [comment]
I merged stuff as far as I thought useful, thanks for all the work. 

Please note that it would make it easier if you can separate out changes some more so they are done with one PR per logical change, this would make it possible to pick certain pieces immediately and discuss others some more before merging them.

**Comment by SUPERCILEX** [comment]
@centic9 Glad to have some of this merged in! To be honest, I completely forgot about this PR which is why there were so many changes in this PR.

**Pull request closed without merging**